apps/meteor/client/views/room/contextualBar/PruneMessages/PruneMessages.tsx

```diff
@@ -20,30 +21,21 @@ type PruneMessagesProps = {
 	callOutText?: string;
 	validateText?: string;
 	users: string[];
-	values: Record<string, unknown>;
-	handlers: Record<string, (eventOrValue: unknown) => void>;
 	onClickClose: () => void;
 	onClickPrune: () => void;
 };
 
-const PruneMessages = ({ callOutText, validateText, values, handlers, onClickClose, onClickPrune }: PruneMessagesProps): ReactElement => {
+const PruneMessages = ({ callOutText, validateText, onClickClose, onClickPrune }: PruneMessagesProps): ReactElement => {
 	const t = useTranslation();
+	const { watch, setValue, control } = useFormContext();
+	const { newerDate, newerTime, olderDate, olderTime } = watch() as typeof initialValues;
 
-	const { newerDate, newerTime, olderDate, olderTime, users, inclusive, pinned, discussion, threads, attached } =
-		values as typeof initialValues;
+	const handleFieldValues = useCallback((field) => (e: any) => setValue(field, e.target.value), [setValue]);
 
-	const {
-		handleNewerDate,
-		handleNewerTime,
-		handleOlderDate,
-		handleOlderTime,
-		handleInclusive,
-		handlePinned,
-		handleDiscussion,
-		handleThreads,
-		handleAttached,
-		handleUsers,
-	} = handlers;
+	const handleNewerDate = handleFieldValues('newerDate');
+	const handleNewerTime = handleFieldValues('newerTime');
+	const handleOlderDate = handleFieldValues('olderDate');
+	const handleOlderTime = handleFieldValues('olderTime');
```

Comment:
`question` Wouldn't be better to change `PruneMessagesDateTimeRow` to accept something more "idiomatic" in RHF? Maybe getting `control` from props or something else.